source/server-discovery-and-monitoring/server-discovery-and-monitoring-tests.rst

```diff
@@ -143,3 +143,29 @@ Command response on an arbiter, recovering member, ghost, or secondary
 when slaveOk is false:
 
     {ok: 0, errmsg: "not master"}
+
+Network error on monitoring connection
```

Comment:
The [SDAM test plan](https://github.com/mongodb/specifications/tree/master/source/server-discovery-and-monitoring/tests#test-phases) already had syntax for simulating a "network error":

> If a response is the empty object {}, simulate a network error.

Is there any reason we need a new prose test for "Non-timeout network error" and cannot instead rely on a spec test? Ideally, we'd also have some way to denote a "network timeout error" via a spec test.

Reply:
@p-mongo: I'm not sure if this question was answered elsewhere in the PR, but can you follow up here? If the SDAM test plan already has syntax for simulating a non-timeout network error, can't we use that instead of introducing prose tests here?

I realize we might still need a prose test for the "network timeout error" case unless you'd rather add new spec test syntax for that.

Reply:
@p-mongo: ping

Reply:
I believe the non-timeout test is now handled via the fail points and events. The timeout test currently specifies the use of mocking, should I create a server ticket requesting a timeout feature for fail points?

Reply:
I'm happy to rely on mocking, rather than make another request for server changes. Mocking also means the tests will conceivably run much quicker without having to wait for an actual timeout (and any flakiness that may entail).